internal/test/daemon/swarm.go

```diff
@@ -85,6 +85,7 @@ func (d *Daemon) SwarmInit(t assert.TestingT, req swarm.InitRequest) {
 		req.DefaultAddrPool = d.DefaultAddrPool
 		req.SubnetSize = d.SubnetSize
 	}
+	req.DataPathPort = d.DataPathPort
```

Comment:
Didn't notice this change really, why was this needed? If no port is specified, won't SwarmKit / libnetwork still set the default automatically? (if not, that would be a problem, because older clients won't set this value).

Reply:
This is for testing new option DataPathPort. You are right,  if  port is not specified , libnetwork will use default value. if no port is specified , here the value ill be 0. also this test is going to run only on API version 40 and above.

Reply:
But this is setting the port now by default, so even if no port is specified in the test, a port is send in the request.

Point is that this utility is used by at least four tests (`TestAPISwarmJoinToken`, `TestUpdateSwarmAddExternalCA`, `TestAPISwarmPromoteDemote`, `TestAPISwarmForceNewCluster`),
and because it's called by `StartAndSwarmInit` (which is called by `AddDaemon` and `NewSwarm`) is used by 130+ other tests, so all of those tests will now change, correct?

Reply:
Hm, so I think my confusion here is that we're overriding the `initRequest` with a daemon configuration; looking a bit further;

So, this PR adds a _daemon_ option `DataPathPort` , so that the port gets automatically set when calling `swarm.NewSwarm()`; https://github.com/moby/moby/blob/5d82d77a6df0c95bb5986ace1255de424c700b33/integration/internal/swarm/service.go#L50-L61


I think the issue lies there;

- Unlike (e.g.) `storageDriver`, `DataPathPort` is not a daemon configuration
- The `DataPathPort` must be given when calling `SwarmInit`

So the problem is that `swarm.NewSwarm()` is doing to much for this test; possible resolutions for that;

- Add a `swarmops ...func(*swarm.InitRequest)` to `swarm.NewSwarm()` (so that the options to initialize the swarm can be passed)
- Don't use `swarm.NewSwarm()` in this test, and first create the daemon, start it (`d.StartNode()`), then call `

```go
d := daemon.New(t)
d.StartNode(t)
d.SwarmInit(t, swarm.InitRequest{DataPathPort: 12345})
```

Reply:
Looks like we've taken the same approach for `DefaultAddrPool`, which also is not a daemon option (but something to set on `swarm init`).

We should fix that, but let's do that in a follow-up